private.php

```diff
@@ -903,7 +903,11 @@
         $db->update_query('privatemessages', $updatearray, "pmid='{$pmid}'");
 
         // Update the unread count - it has now changed.
-        update_pm_count($mybb->user['uid'], 6);
+        $pmcount = update_pm_count($mybb->user['uid'], 6);
+
+		if(is_array($pmcount) && isset($pmcount['unreadpms'])) {
```

Comment:
Indentaion here seems a bit messed up, should be:

```suggestion
    if (is_array($pmcount) && isset($pmcount['unreadpms'])) {
        $mybb->user['pms_unread'] = $pmcount['unreadpms'];
    }
```